msal/application.py

```diff
@@ -299,7 +299,7 @@ def __init__(
         except ValueError:  # Those are explicit authority validation errors
             raise
         except Exception:  # The rest are typically connection errors
-            if validate_authority and region:
+            if validate_authority and azure_region:
```

Comment:
Argh, our last-minute change from `region` to `azure_region` left this one behind. My bad.

Nice catch! Thanks!